File: src/backend/practice/src/main/java/cn/code/practice/controller/IndexController.java
package cn.code.practice.controller;


import cn.code.zeus.common.api.exception.RequestException;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;

import javax.websocket.server.PathParam;


@Controller
public class IndexController {

    @RequestMapping("/index/type")
    public String index( ){

        String type = "4";
        if(type.equals("400"))
            throw new RequestException("400");

        if(type.equals("500"))
            throw new RuntimeException("500");

        return "what fuck";
    }
}

File: src/backend/zeus/component/zookeeper/src/main/kotlin/cn/code/zeus/zk/controller/IndexController.kt
package cn.code.zeus.zk.controller

import org.apache.curator.framework.CuratorFramework
import org.apache.curator.framework.recipes.locks.InterProcessMutex
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/index")
class IndexController {


    @Autowired
    lateinit var client: CuratorFramework

    @GetMapping("/val/{val}")
    fun setValue(
            @PathVariable("val")
            value: String
    ): String? {
        logger.info("try create node")
        client
                .create()
                .creatingParentsIfNeeded()
                .forPath("/usr/default", value.toByteArray())
        val v = client.data.forPath("/usr/default")
        return String(v)
    }

    @GetMapping("/lock/{val}")
    fun lock(
        @PathVariable("val")
        value: String
    ): String? {
        logger.info("try lock for $value")
        val lockPath = "/lock-default"
        val lock = InterProcessMutex(client,lockPath)
        lock.acquire()
        lock.release()
        val v = client.data.forPath("/usr/default")
        return String(v)
    }

    companion object {
        private val logger = LoggerFactory.getLogger(IndexController::class.java)
    }
}